.github/workflows/tf-test.yaml

```diff
@@ -0,0 +1,71 @@
+name: TF Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+env:
+  SPACELIFT_API_KEY_ID: ${{ secrets.SPACELIFT_API_KEY_ID }}
+  SPACELIFT_API_KEY_SECRET: ${{ secrets.SPACELIFT_API_KEY_SECRET }}
+
+permissions:
+  actions: read
+  checks: write
+  contents: read
+  id-token: write
+  pull-requests: read
+
+jobs:
+  terraform-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Aqua Cache
+        uses: actions/cache@v4.0.2
+        if: ${{ !github.event.act }} # Don't enable the cache step if we're using act for testing
+        with:
+          path: ~/.local/share/aquaproj-aqua
+          key: v1-aqua-installer-${{runner.os}}-${{runner.arch}}-${{hashFiles('aqua.yaml')}}
+          restore-keys: |
+            v1-aqua-installer-${{runner.os}}-${{runner.arch}}-
+
+      - name: Install Aqua
+        uses: aquaproj/aqua-installer@4551ec64e21bf0f557c2525135ff0bd2cba40ec7 # v3.0.0
+        with:
+          aqua_version: v2.25.2
+
+      - name: Aqua Install
+        shell: bash
+        run: aqua install --tags terraform
```

Comment:
When reading through this, I was wondering if we should use GitHub Action with the matrix strategy to make it more DRY: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#using-a-matrix-strategy

We could replace any usage of `terraform` or `tofu` with something like ${{ matrix.tooling }}. 

But also at the same time, we're just maintaining these two so it doesn't really matter to make it super optimized.

Reply:
Trying this out!

Reply:
Working and I like it 👍 We can break it up in the future if terraform + tofu decide to diverge in some way that makes this difficult

Reply:
Yep!